src/coinbase/wallet.ts

```diff
@@ -130,31 +136,67 @@ export class Wallet {
   }
 
   /**
-   * Imports a Wallet for the given Wallet data.
+   * Loads an existing CDP Wallet using a wallet data object or mnemonic seed phrase.
    *
-   * @param data - The Wallet data to import.
-   * @param data.walletId - The ID of the Wallet to import.
-   * @param data.seed - The seed to use for the Wallet.
-   * @returns The imported Wallet.
-   * @throws {ArgumentError} If the Wallet ID is not provided.
+   * @param data - The data used to import the wallet:
+   *   - If WalletData: Must contain walletId (or wallet_id) and seed.
+   *     Allows for the loading of an existing CDP wallet into CDP.
+   *   - If MnemonicSeedPhrase: Must contain a valid BIP-39 mnemonic phrase (12, 15, 18, 21, or 24 words).
+   *     Allows for the import of an external wallet into CDP as a 1-of-1 wallet.
+   * @returns A Promise that resolves to the loaded Wallet instance
+   * @throws {ArgumentError} If the data format is invalid.
    * @throws {ArgumentError} If the seed is not provided.
-   * @throws {APIError} If the request fails.
+   * @throws {ArgumentError} If the mnemonic seed phrase is invalid.
    */
-  public static async import(data: WalletData): Promise<Wallet> {
-    if (!data.walletId) {
-      throw new ArgumentError("Wallet ID must be provided");
-    }
-    if (!data.seed) {
-      throw new ArgumentError("Seed must be provided");
+  public static async import(data: WalletData | MnemonicSeedPhrase): Promise<Wallet> {
+    // Check if data is a mnemonic seed phrase object
+    if (isMnemonicSeedPhrase(data)) {
+      // Handle mnemonic seed phrase object import
+
+      if (!data.mnemonicPhrase) {
+        throw new ArgumentError("BIP-39 mnemonic seed phrase must be provided");
+      }
+
+      if (!validateMnemonic(data.mnemonicPhrase, wordlist)) {
+        throw new ArgumentError("Invalid BIP-39 mnemonic seed phrase");
+      }
+
+      // Convert mnemonic phrase to seed
+      const seedBuffer = mnemonicToSeedSync(data.mnemonicPhrase);
+      const seed = hexlify(seedBuffer).slice(2); // remove 0x prefix
```

Comment:
Probably out-of-scope for this PR, but I would love to switch to `viem` and/or [Ox](https://oxlib.sh/).

That said, we do already have `viem` installed as a dependency, so we could use [`toHex`](https://viem.sh/docs/utilities/toHex#tohex) instead!